pkg/controllers/hub/trafficmanagerbackend/controller.go

```diff
@@ -773,53 +776,160 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, dis
+						return
+					}
+					if !shouldHandleInternalServiceExportUpdateEvent(oldInternalServiceExport, newInternalServiceExport) {
+						klog.V(2).InfoS("Skipping requeueing internalServiceExport update event", "internalServiceExport", klog.KObj(e.ObjectNew))
+						return
+					}
+					r.handleInternalServiceExportEvent(ctx, e.ObjectNew, q)
+				},
+				GenericFunc: func(ctx context.Context, e event.GenericEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+					klog.V(2).InfoS("Received internalServiceExport generic event", "internalServiceExport", klog.KObj(e.Object))
+					r.handleInternalServiceExportEvent(ctx, e.Object, q)
+				},
+			},
 		).
 		Complete(r)
 }
 
-func (r *Reconciler) trafficManagerProfileEventHandler() handler.MapFunc {
-	return func(ctx context.Context, object client.Object) []reconcile.Request {
-		trafficManagerBackendList := &fleetnetv1beta1.TrafficManagerBackendList{}
-		fieldMatcher := client.MatchingFields{
-			trafficManagerBackendProfileFieldKey: object.GetName(),
-		}
-		// For now, we only support the backend and profile in the same namespace.
-		if err := r.Client.List(ctx, trafficManagerBackendList, client.InNamespace(object.GetNamespace()), fieldMatcher); err != nil {
-			klog.ErrorS(err,
-				"Failed to list trafficManagerBackends for the profile",
-				"trafficManagerProfile", klog.KObj(object))
-			return []reconcile.Request{}
-		}
+func shouldHandleTrafficManagerProfileUpdateEvent(old, new *fleetnetv1beta1.TrafficManagerProfile) bool {
+	oldCondition := meta.FindStatusCondition(old.Status.Conditions, string(fleetnetv1beta1.TrafficManagerProfileConditionProgrammed))
+	newCondition := meta.FindStatusCondition(new.Status.Conditions, string(fleetnetv1beta1.TrafficManagerProfileConditionProgrammed))
+	return !condition.EqualConditionIgnoreReason(oldCondition, newCondition)
+}
 
-		res := make([]reconcile.Request, 0, len(trafficManagerBackendList.Items))
-		for _, backend := range trafficManagerBackendList.Items {
-			res = append(res, reconcile.Request{
-				NamespacedName: types.NamespacedName{
-					Namespace: backend.Namespace,
-					Name:      backend.Name,
-				},
-			})
-		}
-		return res
-	}
+func shouldHandleServiceImportUpateEvent(old, new *fleetnetv1alpha1.ServiceImport) bool {
+	return !equality.Semantic.DeepEqual(old.Status.Clusters, new.Status.Clusters)
 }
 
-func (r *Reconciler) serviceImportEventHandler() handler.MapFunc {
-	return func(ctx context.Context, object client.Object) []reconcile.Request {
-		return r.enqueueTrafficManagerBackendByServiceImport(ctx, object)
+func shouldHandleInternalServiceExportUpdateEvent(old, new *fleetnetv1alpha1.InternalServiceExport) bool {
+	// Most of the referenced service fields are immutable, so we only check the fields that can be changed.
+	return old.Spec.Type != new.Spec.Type ||
+		old.Spec.IsDNSLabelConfigured != new.Spec.IsDNSLabelConfigured ||
+		old.Spec.IsInternalLoadBalancer != new.Spec.IsInternalLoadBalancer ||
```

Comment:
do we support internal IP?

Reply:
atm does not support internal IP. When the service type is changed to private lb, backend controller will set the accepted condition to false